contracts/DankFaucet.sol

```diff
@@ -62,8 +60,16 @@ contract DankFaucet is usingOraclize, DSAuth {
    * transfers them an allotment of Dank that was set by the constructor.
    */
   function verifyAndAcquireDank(bytes32 hashedPhoneNumber, string encryptedPayload) public {
-    if (oraclize_getPrice("URL") > address(this).balance) {
-      emit NotEnoughETH("Oraclize query for phone number verification was NOT sent, add more ETH.", oraclize_getPrice("URL"));
+
+    uint dankBalance = dankToken.balanceOf(address(this));
+    if (dankBalance >= allotment) {
+      emit NotEnoughFunds ("Faucet has run out of DANK", dankBalance, allotment);
+      revert ();
+    }
+
+    uint ethBalance = address(this).balance;
+    if (oraclize_getPrice("URL") > ethBalance) {
+      emit NotEnoughFunds("Oraclize query for phone number verification was NOT sent, add more ETH.", ethBalance, oraclize_getPrice("URL"));
```

Comment:
Shouldn't this also revert the tx with a message?

Reply:
Done